test/ng/directive/inputSpec.js

```diff
@@ -1941,6 +1977,111 @@ describe('input', function() {
+        ['2010-01-41T00:00:00.0000+01:01', false],  // day value too large
+        ['2010-01-200T00:00:00.0000+01:01', false], // day value too many digits
+
+        // Validate time
+        ['2010-01-01', false],                      // time must be specified
+        ['2010-01-0101:00:00.0000+01:01', false],   // missing date time seperator
+        ['2010-01-01V01:00:00.0000+01:01', false],  // invalid date time seperator
+        ['2010-01-01T01-00-00.0000+01:01', false],  // time must use period seperator
+
+        // Validate hour
+        ['2010-01-01T01:00:00.0000+01:01', true],
+        ['2010-01-01T:00:00.0000+01:01', false],    // hour value must be specified
+        ['2010-01-01T1:00:00.0000+01:01', false],   // hour value too few digits
+        ['2010-01-01T-01:00:00.0000+01:01', false], // hour value must be positive
+        ['2010-01-01T32:00:00.0000+01:01', false],  // hour value too large
+        ['2010-01-01T220:00:00.0000+01:01', false], // hour value too many digits
+
+        // Validate minutes
+        ['2010-01-01T01:00:00.0000+01:01', true],
+        ['2010-01-01T01::00.0000+01:01', false],    // minute value must be specified
+        ['2010-01-01T01:0:00.0000+01:01', false],   // minute value too few digits
+        ['2010-01-01T01:-00:00.0000+01:01', false], // minute value must be positive
+        ['2010-01-01T01:60:00.0000+01:01', false],  // minute value too large
+        ['2010-01-01T01:100:00.0000+01:01', false], // minute value too many digits
+
+        // Validate seconds
+        ['2010-01-01T01:00:00.0000+01:01', true],
+        ['2010-01-01T01:00:.0000+01:01', false],    // second value must be specified
+        ['2010-01-01T01:00:0.0000+01:01', false],   // second value too few digits
+        ['2010-01-01T01:00:-00.0000+01:01', false], // second value must be positive
+        ['2010-01-01T01:00:60.0000+01:01', false],  // second value too large
+        ['2010-01-01T01:00:100.0000+01:01', false], // second value too many digits
+
+        // Validate milliseconds
+        ['2010-01-01T01:00:00+01:01', false],       // millisecond value must be specified
+        ['2010-01-01T01:00:00.+01:01', false],      // millisecond value must be specified
+        ['2010-01-01T01:00:00.-0000+01:01', false], // millisecond value must be positive
+
+        // Validate timezone
+        ['20123-06-15T00:00:00.0000', false],       // timezone must be specified
+
+        // Validate timezone offset
+        ['2010-06-15T00:00:00.0000+01:01', true],
+        ['2010-06-15T00:00:00.0000-01:01', true],
+        ['2010-06-15T00:00:00.0000~01:01', false],  // timezone must have valid postive/negative indicator
+        ['2010-06-15T00:00:00.000001:01', false],   // timezone must have valid postive/negative indicator
+        ['2010-06-15T00:00:00.0000+00:01Z', false], // timezone invalid trailing characters
+        ['2010-06-15T00:00:00.0000+00:01 ', false], // timezone invalid trailing characters
```

Comment:
Actually, according to the standard, the timezone offset can have one of three forms:

`+/-HH:mm`, `+/-HHmm`, `+/-HH`

(Feel free to add suppory for those as well :smiley:)